devel/zthread/Portfile

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+
+name                zthread
+version             2.3.2
+categories          devel
+license             MIT
+maintainers         nomaintainer
+                    #  v Note: No https site available
+homepage            http://${name}.sourceforge.net
+description         object-oriented threading and synchronization \
+                    library
+long_description    ZThread is an advanced ${description}, implemented \
+                    in C++. It provides an excellent and powerful \
+                    abstraction from native threads. It includes \
+                    interruptible thread objects and several other \
+                    synchronization control objects and lets you \
+                    safely terminate threads without the complications \
+                    of using cancellation handlers or other similar \
+                    constructs.
+
+master_sites        sourceforge:project/${name}/ZThread/${version}
+distname            ZThread-${version}
+checksums           rmd160  8419afad91ea8db1fff7ce3bf7784329839dd2e8 \
+                    sha256  950908b7473ac10abb046bd1d75acb5934344e302db38c2225b7a90bd1eda854 \
+                    size    412527
+
+post-patch {
+    foreach re [list \
+        {/void createScope.*long ms/s/void/bool/} \
+        {/if.*isDisabled/s/(isDisabled)/LockHolder<LockType>::\1/} \
+        {/LockingPolicy.*Scope.*extract/s/(extract)/this->\1/} \
+    ] {
+        reinplace -E $re ${worksrcpath}/include/zthread/Guard.h
+    }
+    foreach re [list \
+        {/waiter(Arrived|Departed).*self/s/(waiter(Arrived|Departed))/MutexImpl<List, Behavior>::\1/} \
+        {/owner(Acquired|Released).*(self|impl)/s/(owner(Acquired|Released))/MutexImpl<List, Behavior>::\1/} \
+    ] {
+        reinplace -E $re ${worksrcpath}/src/MutexImpl.h
+    }
+
+    # Incorporate bugfix from Debian package:
+    # "030_executor-remove-thread-bug"
+    reinplace -E \
+        {/remove.*threads.*current/s/(std::remove.*);/_threads.erase(\1);/} \
+        ${worksrcpath}/src/PoolExecutor.cxx
+
+    # Incorporate bugfix from Debian package: "080_wrong-parameter-type"
+    reinplace {/virtual bool add/s/T item/const T\& item/} \
+        ${worksrcpath}/include/${name}/BlockingQueue.h
+
+    # Add missing DESTDIR to some of the install paths
+    foreach re [list \
+        {/mkinstalldirs.*(bin|data)dir/s/(\$.(bin|data)dir)/\$(DESTDIR)\1/} \
+        {/INSTALL_PROGRAM.*zthread-config/s/(\$.bindir)/\$(DESTDIR)\1/} \
+        {/INSTALL_DATA.*datadir.*aclocal/s/(\$.datadir)/\$(DESTDIR)\1/} \
+    ] {
+        foreach f [list Makefile.am Makefile.in] {
+            reinplace -E $re ${worksrcpath}/$f
+        }
+    }
+}
```

Comment:
all of this should (probably) be done with patchfiles instead of incomprehensible reinplaces...

Reply:
Believe it or not, using patchfiles would actually be a lot messier, because there are quite a large number of (identical) replaces being done.

Reply:
A large number of identical replacements is a good candidate for `reinplace`. However the replacements you tend to write into your ports are complex and it would not surprise me if nobody but you understands what you're changing and why, and nobody but you will have any interest in understanding them if it should be come necessary to change them in the future.

A failed `reinplace` [does not cause the build to stop](https://trac.macports.org/ticket/60844). This makes it easy for you or someone else attempting to update a port to neglect to fix or remove a `reinplace` if the upstream source has changed the code that's being changed.

References to upstream pull requests or bug reports should be added so that we can check there if the fixes are still needed in the future.